pkg/driver/controller.go

```diff
@@ -705,6 +722,10 @@ func (cs *controller) CreateSnapshot(
 	if err != nil {
 		return nil, err
 	}
+	unlockVol := cs.LockVolume(volumeID)
```

Comment:
ok for now but if we have more uses of the volume&snapshot, might be worth having a function to lock, ensuring the locks are always taken in the same order to avoid deadlock